src/Symfony/Component/Cache/DependencyInjection/CachePoolPass.php

```diff
@@ -0,0 +1,150 @@
+        } else {
+            $seed = '_'.$container->getParameter('kernel.root_dir');
+        }
+        $seed .= '.'.$container->getParameter('kernel.name').'.'.$container->getParameter('kernel.environment');
+
+        $pools = array();
+        $clearers = array();
+        $attributes = array(
+            'provider',
+            'name',
+            'namespace',
+            'default_lifetime',
+            'reset',
+        );
+        foreach ($container->findTaggedServiceIds('cache.pool') as $id => $tags) {
+            $adapter = $pool = $container->getDefinition($id);
+            if ($pool->isAbstract()) {
+                continue;
+            }
+            while ($adapter instanceof ChildDefinition) {
+                $adapter = $container->findDefinition($adapter->getParent());
+                if ($t = $adapter->getTag('cache.pool')) {
+                    $tags[0] += $t[0];
+                }
+            }
+            $name = $tags[0]['name'] ?? $id;
+            if (!isset($tags[0]['namespace'])) {
+                $tags[0]['namespace'] = $this->getNamespace($seed, $name);
+            }
+            if (isset($tags[0]['clearer'])) {
+                $clearer = $tags[0]['clearer'];
+                while ($container->hasAlias($clearer)) {
+                    $clearer = (string) $container->getAlias($clearer);
+                }
+            } else {
+                $clearer = null;
+            }
+            unset($tags[0]['clearer'], $tags[0]['name']);
+
+            if (isset($tags[0]['provider'])) {
+                $tags[0]['provider'] = new Reference(static::getServiceProvider($container, $tags[0]['provider']));
+            }
+            $i = 0;
+            foreach ($attributes as $attr) {
+                if (!isset($tags[0][$attr])) {
+                    // no-op
+                } elseif ('reset' === $attr) {
+                    if ($tags[0][$attr]) {
+                        $pool->addTag('kernel.reset', array('method' => $tags[0][$attr]));
+                    }
+                } elseif ('namespace' !== $attr || ArrayAdapter::class !== $adapter->getClass()) {
+                    $pool->replaceArgument($i++, $tags[0][$attr]);
+                }
+                unset($tags[0][$attr]);
+            }
+            if (!empty($tags[0])) {
+                throw new InvalidArgumentException(sprintf('Invalid "cache.pool" tag for service "%s": accepted attributes are "clearer", "provider", "name", "namespace", "default_lifetime" and "reset", found "%s".', $id, implode('", "', array_keys($tags[0]))));
+            }
+
+            if (null !== $clearer) {
+                $clearers[$clearer][$name] = new Reference($id, $container::IGNORE_ON_UNINITIALIZED_REFERENCE);
+            }
+
+            $pools[$name] = new Reference($id, $container::IGNORE_ON_UNINITIALIZED_REFERENCE);
+        }
+
+        $clearer = 'cache.global_clearer';
```

Comment:
we moving to DI passes to component, we usually try to hardcode as little identifiers as possible, using a constructor with default arguments (see e.g. `CachePoolPrunerPass`)